src/main/java/org/veupathdb/service/eda/ds/plugin/AbstractPlugin.java

```diff
@@ -141,8 +141,9 @@ public final AbstractPlugin<T,S,R> processRequest(String appName, T request, Ent
       throw new BadRequestException("Compute results are not available for the requested job.");
     }
 
-    // if plugin requires a compute, get computed var metadata
-    List<VariableMapping> computedVars = _computeInfo.isPresent()
+    // if plugin requires a compute and metadata is expected for this compute, get computed var metadata
+    boolean expectMetadata = !appName.equals("differentialabundance");  // Diff abundance has no computed var metadata
```

Comment:
I think there's a better way to do this- we could either have the plugin declare whether they expect to have var metadata (always false for non-compute plugins) or we can handle the 404 and return emptyList() in that case.  The first way forces the plugins to override a method if they have computes that don't provide vars; the second way means we're making that service call unnecessarily just to get a 404 back.  So I think the first way is probably better.

Reply:
Also if the data plugin thinks there should be vars but we get a 404 that should throw an error to alert someone to the mismatch.

Reply:
Added here: https://github.com/VEuPathDB/EdaDataService/pull/267/commits/d8862e724789689d8c59858a4d4d66c1d102e1ff

Reply:
Thanks @ryanrdoherty ! That makes a lot of sense